libsentrykube/service.py

```diff
@@ -112,9 +124,109 @@ def get_service_value_overrides(
 
         with open(service_override_file, "rb") as f:
             values = yaml.safe_load(f)
+
+        # Try to load the config only after {cluster_name}.yaml was successfully loaded because the common override
+        # only makes sense if there is a cluster config
+        common_override_values = get_common_regional_override(
+            service_name, region_name, external
+        )
+
+        if common_override_values:
+            deep_merge_dict(common_override_values, values)
+            return common_override_values
+
+        return values
     except FileNotFoundError:
-        values = {}
-    return values
+        return {}
+
+
+def get_common_regional_override(
+    service_name: str, region_name: str, external: bool = False
+) -> dict:
+    """
+    Helper function to load common regional configuration values.
+
+    Looks for a '_values.yaml' file in the region's override directory that contains
+    settings shared across all clusters in that region.
+    """
+    try:
+        common_service_override_file = (
+            get_service_value_override_path(service_name, region_name, external)
+            / "_values.yaml"
+        )
+
+        with open(common_service_override_file, "rb") as f:
+            common_override_values = yaml.safe_load(f)
+
+        return common_override_values
+    except FileNotFoundError:
+        return {}
+
+
+def get_hierarchical_value_overrides(
+    service_name: str,
+    region_name: str,
+    cluster_name: str = "default",
+    external: bool = False,
+) -> dict:
+    """
+    Enables hierarchical configuration overrides with shared base values.
+
+    This function extends the standard region_overrides system by adding support for
+    shared base configurations. This helps reduce duplication across region-specific
+    configurations.
+
+    Directory Structure:
+        region_overrides/
+        └── common_shared_config/      # Arbitrary name for the shared config group
+            ├── _values.yaml           # Base values for this group
+            └── {region_name}/         # Region-specific overrides
+                └── {cluster_name}.yaml # Cluster-specific overrides
```

Comment:
I am not sure I follow the reasoning here.
1) Which scenario do you want to use the hierarchical value overrides in? Is it about single tenant in order to have some common config for all single tenants vs config common to all saas regions? While I hate the idea that ST should be different I guess that's the world we live into. But if this is not targeting that scenario, could you please elaborate ?
2) If you want to provide a value file that is common to a group of regions why do you have a cluster file here as well? I think in that case the standard cluster specific file (`/region_override/my_cluster.yaml`) should apply rather than having a separate one. In other word I would like this cross regions defaults to apply within the same hierarchy structure we use for all the other services rather than having two parallel hierarchies.

Reply:
> Which scenario do you want to use the hierarchical value overrides in? Is it about single tenant in order to have some common config for all single tenants vs config common to all saas regions? While I hate the idea that ST should be different I guess that's the world we live into. But if this is not targeting that scenario, could you please elaborate ?

Yes it's for single tenants. I agree that it's not great that the config diverges

> If you want to provide a value file that is common to a group of regions why do you have a cluster file here as well? I think in that case the standard cluster specific file (/region_override/my_cluster.yaml) should apply rather than having a separate one. In other word I would like this cross regions defaults to apply within the same hierarchy structure we use for all the other services rather than having two parallel hierarchies.

Consider the following structure:
```
region_overrides/
├── customer1/
├── customer2/
├── customer3/
├── customer4/
└── customer5/
``` 
`customer1` and `customer2` have similar config and `customer3`, `customer4` and `customer5` have also similar config with enough differences that they can't share a common config with `customer1` and `customer2`. 
As far as I know there is no support for this type of structure which results in a lot of copied config in either the first two or the last 3 customer configs.
Even if we would use `/region_override/my_cluster.yaml`, we can still only share common configs for either of those groups and the other one needs to overwrite it per cluster file.
With my changes we could split it into the following structure where each group can have a single file with a common config:
```
region_overrides/
├── first-group/
│   ├── customer1/
│   └── customer2/
└── second-group/
    ├── customer3/
    ├── customer4/
    └── customer5/
```
Additionally to reducing code duplication we would also see differences between customers more easily.

Reply:
I'd be fine in having groups. 
Though the override systems must enforce that a given region/cluster must be defined only once. 
Either `us/default.yaml` is in a group or it is in the root `region_override`. It is critical we do not allow both at the same time.


Also please psot this PR in #team-sre-squad-monkeys as this will be an important change that impacts all services and those who manage all services may have opinions.